src/Core/IThreadHandling.cs

```diff
@@ -61,8 +61,18 @@ public interface IThreadHandling
         /// If the caller is not on the main thread then the method will switch to the main thread,
         /// then resume on the caller's thread when then the operation completes.
         /// </summary>
+        [Obsolete]
```

Comment:
Could you add a comment explaining why this is obsolete and what the alternative is, please? e.g.

```
[Obsolete("This method does not work correctly. Use RunOnUIThreadSync2 instead.")]
// See https://github.com/SonarSource/sonarlint-visualstudio/issues/4179 for more info.
```

I think the message text in the attribute might show up in IntelliSense, but I'm not sure.